### PR TITLE
querydb: add android query for fragment injection

### PR DESCRIPTION
**Pull request by ursachec**

### CODE REVIEW

File: querydb/src/main/scala/io/joern/scanners/android/UnsafeReflection.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        val targetSdkVersionMatch                                   = """^[^t]+targetSdk[^0-9]+(\d+)""".r
+        val minimumAndroidSdkVersionWhereNoAdditionalChecksRequired = 19
+        groovyBuildGradleFiles.filter { gradleFile =>
+          gradleFile.content
+            .split('\n')
+            .filter { line =>
+              targetSdkVersionMatch
+                .findAllIn(line)
+                .matchData
+                .toList
```
Comment (maltek) [inline querydb/src/main/scala/io/joern/scanners/android/UnsafeReflection.scala:39]:
```suggestion
```

I expect we can just drop this line?

File: querydb/src/main/scala/io/joern/scanners/android/UnsafeReflection.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                .filter { m =>
+                  if (m.groupCount > 0) m.group(1).toInt < minimumAndroidSdkVersionWhereNoAdditionalChecksRequired
```
Comment (maltek) [inline querydb/src/main/scala/io/joern/scanners/android/UnsafeReflection.scala:41]:
```suggestion
                  m.groupCount > 0 && m.group(1).toInt < minimumAndroidSdkVersionWhereNoAdditionalChecksRequired
```

File: querydb/src/main/scala/io/joern/scanners/android/UnsafeReflection.scala [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                  else false
+                }
+                .nonEmpty
+            }
+            .nonEmpty
+        }.l
```
Comment (maltek) [inline querydb/src/main/scala/io/joern/scanners/android/UnsafeReflection.scala:47]:
```suggestion
        }
```

**Review by maltek — APPROVED** [review]

**Comment by ursachec** [comment]
yess to all, thx & done

**Pull request merged**